monai/data/thread_buffer.py

```diff
@@ -106,21 +168,30 @@ class ThreadDataLoader(DataLoader):
         buffer_size: number of items to buffer from the data source.
         buffer_timeout: time to wait for an item from the buffer, or to wait while the buffer is full when adding items.
         repeats: number of times to yield the same batch.
+        use_thread_workers: if True and num_workers > 0 the workers are created as threads instead of processes
         kwargs: other arguments for `DataLoader` except for `dataset`.
 
     """
 
     def __init__(
-        self, dataset: Dataset, buffer_size: int = 1, buffer_timeout: float = 0.01, repeats: int = 1, **kwargs
+        self,
+        dataset: Dataset,
+        buffer_size: int = 1,
+        buffer_timeout: float = 0.01,
+        repeats: int = 1,
+        use_thread_workers: bool = False,
```

Comment:
Hi @ericspod ,

Should we default this value to `True` directly?
BTW, do we still have the thread-safe issue about random transforms here?
For example: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L525
If true, I think at least we should describe it in the doc-string?

Thanks.

Reply:
We do have the thread-safety or just thread-weirdness issues when threads are used, so I would definitely leave it as False for now. The advantage of threads over processes isn't always there anyway, it depends on the transforms being used.